### PR TITLE
Deprecate all usages of chained option setters.

### LINKED ISSUE #396: Deprecate all usages of chained option setters
Options should be set functionally like this:

```go
type RequestOption func(*Request)

func NewHTTPRequest(req *http.Request, opts ...RequestOption) *Request {
	hreq := &Request{http: req}

	for _, opt := range opts {
		opt(hreq)
	}

	return hreq
}

func WithAuth(auth func(*http.Request) (*http.Response, error)) RequestOption {
	return func(req *Request) {
		req.auth = auth
	}
}
```

Remove optional setter methods from the HTTPService and replace them with functional options.

### PR DESCRIPTION
**Pull request by Conor-Fleming**
Resolves #396 

Let me know what you think or if I have missed any instances of chained option setters.

### CODE REVIEW

File: http_test.go [hunk 1]
```diff
@@ -125,10 +125,15 @@ func TestIterator(t *testing.T) {
 						errReq = reqs[0]
 					}
```
Comment (Conor-Fleming) [inline http_test.go:127]: I wasnt sure how best to format this code but this way looked the cleanest to me. Let me know what you think and if youd like any changes. @prestonvasquez

File: http_test.go [hunk 2]
```diff
@@ -125,10 +125,15 @@ func TestIterator(t *testing.T) {
...
-						withMockHTTPClientRequests(reqs...))
+					httpSvc := NewHTTPService(
+						svc,
+						// options
+						WithRateLimiter(rlimiter),
+						WithRequests(reqs...),
+						WithClient(newMockHTTPClient(
+							withMockHTTPClientResponseError(errReq, tcase.forceErr),
+							withMockHTTPClientRequests(reqs...)),
+						))
```
Comment (Conor-Fleming) [inline http_test.go:137]: I wasnt sure how best to format this code but this way looked the cleanest to me. Let me know what you think and if youd like any changes. @prestonvasquez

**Comment by prestonvasquez** [comment]
@Conor-Fleming The code looks good, but seeing the examples I think it might be best to use the chaining method for now. I don't love the idea of having to construct a new http service using the general service like this:

```go
httpSvc := gidari.NewHTTPService(
  svc,
  gidari.WithRequests(charReqWrapper, housReqWrapper),
  gidari.WithRateLimiter(rate.NewLimiter(rate.Every(1*time.Second), 5)))
```

This is my fault, my apologies. I think there might be a use case to support both in the future. But sourcing HTTP through the general service feels more natural, like this:

```go
svc.HTTP.Requests(charReqWrapper, housReqWrapper)
```

What are your thoughts?

**Comment by Conor-Fleming** [comment]
@prestonvasquez Yeah I think I'd have to agree. When writing it, it did seem more natural and more clear, at least in this instance.

**Comment by Conor-Fleming** [comment]
@prestonvasquez do you want me to go ahead and close this PR?

**Comment by prestonvasquez** [comment]
@Conor-Fleming Yes, I think the current solution works for now. Apologies for the inconvenience and thank you for looking at this issue.

**Comment by Conor-Fleming** [comment]
@prestonvasquez Ah no bother at all, this is all helpful for me to learn either way. Thanks!

**Pull request closed without merging**